src/components/reportPage/UploadBox.jsx

```diff
@@ -2,44 +2,128 @@ import React, { useCallback, useEffect, useState } from "react";
+    } else {
+      let errorList = {
+        errorFiles: {
+          overMaxSize: [],
+          duplicateFiles: [],
+          invalidType: [],
+        },
+        errorMsgs: {
+          overMaxSize: `The following file(s) are larger than ${maxFileSize} MB: `,
+          duplicateFiles: "The following file(s) have already been uploaded:",
+          invalidType: INVALID_FILETYPE,
+        },
+      };
+      let errorFiles = errorList.errorFiles;
+      for (let f in fileRejections) {
+        const { errors, file } = fileRejections[f];
+        let errorType = errors[0].code;
+        if (errorType === "file-invalid-type") {
+          errorFiles.invalidType = [...errorFiles.invalidType, file.name];
+        } else if (errorType === "duplicate-file") {
+          errorFiles.duplicateFiles = [...errorFiles.duplicateFiles, file.name];
+        } else if (errorType === "file-too-large") {
+          errorFiles.overMaxSize = [...errorFiles.overMaxSize, file.name];
+        }
       }
+
+      let msg = createErrorMsg(errorList);
+      setAlert(msg);
     }
   };
 
-  // Cusomter file validation function
+  // @param: errorObj should object composed of two objs: errorFiles, errorMsgs, both have identical property names
+  const createErrorMsg = (errorObj) => {
+    let { errorFiles, errorMsgs } = errorObj;
+    let errorTypes = Object.keys(errorFiles);
+    const errors = errorTypes.map((type) =>
+      errorFiles[type].length ? (
+        <>
+          <p style={{ marginBottom: "0.2rem" }}>{errorMsgs[type]}</p>{" "}
+          <ul>
+            {" "}
+            {errorFiles[type].map((file) => (
+              <li key={file}>{file}</li>
+            ))}
+          </ul>{" "}
+        </>
+      ) : null
+    );
+    const errorStyle = { marginLeft: "1rem", marginRight: "1.5rem", overflowWrap: "break-word" };
+    const errorHeadingStyle = { marginBottom: "0.5rem" };
+    let msg = (
+      <>
+        <p style={errorHeadingStyle}>
+          {" "}
+          <b> Rejected Files: </b>{" "}
+        </p>
+        <div style={errorStyle}>{errors}</div>
+      </>
+    );
+    return msg;
+  };
+
+  // Customer file validation function
   const validateFiles = (file) => {
-    if (!props.currFiles.every((currFile) => currFile.name != file.name)) {
-      return {};
+    // Tests if can accept more files
+    if (currFiles.length == fileLimit) {
+      return { code: "over-max-filelimit" };
     }
+
+    // Tests duplicate files
+    else if (!props.currFiles.every((currFile) => currFile.name != file.name)) {
+      return { code: "duplicate-file" };
+    }
+
     // Returns null if files should be accepeted
     return null;
   };
+
   const acceptedFileTypes = ["image/jpeg", "image/png", "video/mp4"];
-  const maxFiles = props.fileLimit == Infinity ? 0 : props.fileLimit;
+  const maxFiles = fileLimit ? fileLimit : 0; // 0 unlimited files
+  const maxSize = maxFileSize ? maxFileSize * 1024 * 1024 : 0; // 0 is unlimited file size, in bin bytes
```

Comment:
I was able to upload a file which was 6.1 MB
This should be `maxFileSize * 1000 * 1000`

Reply:
This should not matter because different OS use different measurements for Megabytes, for example windows calculates a megabyte to be equal to 1024*1024 while ubuntu uses 1000*1000. Its best if we stick with 1024, since this ensures that windows users will not have a problem. I'm assuming that the limit is not a backend requirement/constraint.

Reply:
The limit is a backend constraint